### DISCUSSION TITLE
Union two models in new model.

### DISCUSSION THREAD

**Discussion by Zombach** [riok/mapperly#989] [Q&A] [answered]
Is it possible to combine two models into one? Ignoring empty properties from the second model
```
//Base
public class A
{
      public string? V {get;set;} = "123";
      public string? V2 {get;set;} = "text";
}

//Adding
public class A
{
      public string? V {get;set;} = null;
      public string? V2 {get;set;} = "111";
}

//Result
public class A
{
      public string V {get;set;} = "123";
      public string V2 {get;set;} = "111";
}
```

**Comment by TimothyMakkison** [comment]
Have you tried using [existing target mappings](https://mapperly.riok.app/docs/configuration/existing-target/). Not sure about the nullability behaviour.

**Comment by latonz** [comment, marked as answer]
You can use [existing target mappings](https://mapperly.riok.app/docs/configuration/existing-target/) (as mentioned by @TimothyMakkison) in combination with [`AllowNullPropertyAssignment = false`](https://mapperly.riok.app/docs/configuration/mapper/#null-values), this should work to merge objects where all source properties are nullable.

```c#
[Mapper(AllowNullPropertyAssignment = false)]
static partial class Mapper
{
    public static partial void Merge(A source, B target);
}

// generates
public static partial void Merge(global::A source, global::B target)
{
    if (source.V != null)
    {
        target.V = source.V;
    }
    if (source.V2 != null)
    {
        target.V2 = source.V2;
    }
}
```
